pkg/controller/nodenetworkstatereport/nodenetworkstatereport_controller.go

```diff
@@ -0,0 +1,128 @@
+package nodenetworkstatereport
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	k8shandler "sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	nmstate "github.com/nmstate/kubernetes-nmstate/pkg/helper"
+)
+
+var (
+	log                     = logf.Log.WithName("controller_nodenetworkstatereport")
+	nodenetworkstateRefresh time.Duration
+)
+
+func init() {
+	refreshTime, isSet := os.LookupEnv("NODE_NETWORK_STATE_REFRESH_INTERVAL")
+	if !isSet {
+		panic("NODE_NETWORK_STATE_REFRESH_INTERVAL is mandatory")
+	}
+	intRefreshTime, err := strconv.Atoi(refreshTime)
+	if err != nil {
+		panic(fmt.Sprintf("Failed while converting evnironment variable to int: %v", err))
+	}
+	nodenetworkstateRefresh = time.Duration(intRefreshTime) * time.Second
+}
+
+// Add creates a new NodeNetworkState Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileNodeNetworkStateReport{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("nodenetworkstatereport-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	forThisNode := predicate.Funcs{
+		CreateFunc: func(createEvent event.CreateEvent) bool {
+			return nmstate.EventIsForThisNode(createEvent.Meta)
+		},
+		DeleteFunc: func(deleteEvent event.DeleteEvent) bool {
+			// This controller responsability is updates, receiving
+			// deletes is of no use
+			return false
+		},
+		UpdateFunc: func(updateEvent event.UpdateEvent) bool {
```

Comment:
Just an idea, in order to rereconcile right after a configuration is set, we could write `something somewhere` in the CR once saved (maybe version is already doing that), that change will be observed by report controller and status refreshed. I guess it happens even now, not sure

Reply:
The only "problem" we have with this is that in case desiredState is apply before our Reconcile here at update we will have to wait for next run... I don't see this as a big problem.